### PR TITLE
Add additional logging to Amplitude plugin

### PR DESCRIPTION
**Pull request by willbarrett**
@change/data-science @change/devops [JIRA](https://change.atlassian.net/browse/CHANGE-37285)

What? Add logging and statsd metric emission to the Amplitude plugin
Why? So we can have a clearer understanding of what the plugin is doing at any given time.

Looking for comments on the general idea - if things look good, I'll add a few more specs before merge & version bump.

### CODE REVIEW

File: lib/fluent/plugin/out_amplitude.rb [hunk 1]
```diff
@@ -163,15 +166,27 @@ def extract_event_properties!(amplitude_hash, record)
...
+          records_sent += records_to_send.length
+          log.info(
+            "Amplitude request complete. Duration: #{res.total_time * 1000}, Records: #{records_to_send.length}"
+          )
+        else
           fail_time = Time.now.to_i
           errors << [res.response_code, res.body, records_to_send, fail_time - start_time]
         end
       end
+      log.info("sent #{records_sent} to amplitude")
```
Comment (joycechangeorg) [inline lib/fluent/plugin/out_amplitude.rb]: can we also log this to statsd?

Comment (willbarrett) [reply lib/fluent/plugin/out_amplitude.rb]: That would double-track, since we're tracking after each batch on line 178.

**Comment by joycechangeorg** [comment]
this is awesome

File: lib/fluent/plugin/out_amplitude.rb (hunk 1 above)
Comment (joycechangeorg) [reply lib/fluent/plugin/out_amplitude.rb]: ok, sorry i see it now, so maybe track errors, too?

**Comment by kamalama** [comment]
lgtm

**Pull request merged**